src/posix/platform/max_power_table.hpp

```diff
@@ -0,0 +1,93 @@
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef OT_POSIX_PLATFORM_MAX_POWER_TABLE_HPP_
+#define OT_POSIX_PLATFORM_MAX_POWER_TABLE_HPP_
+
+#include "core/radio/radio.hpp"
+
+namespace ot {
+namespace Posix {
+
+class MaxPowerTable
+{
+public:
+    static const int8_t kPowerDefault = 30; ///< Default power 1 watt (30 dBm).
+
+    MaxPowerTable(void) { memset(mPowerTable, kPowerForbidden, sizeof(mPowerTable)); }
+
+    /**
+     * This method gets the max allowed transmit power of channel @p aChannel.
+     *
+     * @params[in]  aChannel    The radio channel number.
+     *
+     * @returns The max allowed transmit power in dBm.
+     *
+     */
+    int8_t GetTransmitPower(uint8_t aChannel) const { return mPowerTable[aChannel - Radio::kChannelMin]; }
+
+    /**
+     * This method sets the max allowed transmit power of channel @p aChannel.
+     *
+     * @params[in]  aChannel    The radio channel number.
+     * @params[in]  aPower      The max allowed transmit power in dBm.
+     *
+     */
+    void SetTransmitPower(uint8_t aChannel, int8_t aPower) { mPowerTable[aChannel - Radio::kChannelMin] = aPower; }
+
+    /**
+     * This method gets the allowed channel masks.
+     *
+     * All channels of max power value of 0x7f is considered forbidden.
+     *
+     */
+    uint32_t GetAllowedChannelMask(void) const
+    {
+        uint32_t channelMask = 0;
+
+        for (uint8_t i = Radio::kChannelMin; i <= Radio::kChannelMax; ++i)
+        {
+            if (mPowerTable[i - Radio::kChannelMin] != kPowerForbidden)
+            {
+                channelMask |= (1 << i);
+            }
+        }
+
+        return channelMask;
+    }
+
+private:
+    static const int8_t kPowerForbidden = 0x7f;
```

Comment:
Why not enum, which provides greater flexibility in its use (e.g. switch statements, static variable initializer, etc.)?